extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/SceneDiscoveryServiceManager.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.digitalstrom.internal.discovery;
+
+import java.util.HashMap;
+
+import org.eclipse.smarthome.binding.digitalstrom.handler.BridgeHandler;
+import org.eclipse.smarthome.binding.digitalstrom.handler.SceneHandler;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.listener.SceneStatusListener;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.scene.InternalScene;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+
+/**
+ * The {@link SceneDiscoveryServiceManager} manages the different scene discovery services and informs them about new
+ * added scenes.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ */
+public class SceneDiscoveryServiceManager implements SceneStatusListener {
```

Comment:
We implemented the SceneDiscoveryServiceManager afterwards to provide the feature, that every user can decide by himself which kind of scene-types will be discoverd. The user can choose between apartement, zone, group and named scenes. The gaol was to don't flood the inbox with to much scenes.

Before there was only one SceneDiscoveryService for all scene-types, which implemented a SceneListener and registered it to the SceneManager to get informed by new or deleted scenes. 

So it was the easiest way to create the SceneDiscoveryServiceManager, which registers as SceneListener to the SceneManager to get informed about new or deleted scenes and forward this information to the respective SceneDiscoveryServices. Otherwise we had to change the SceneManager etc. in the way, that they can register a SceneListener for each scene-type and that would be much more costlier.

Another reason was, that the implementation of the SceneDiscovery  is equal until the scene-type. Therefore, the scene-type is passed via the constructor and thus creates a SceneDiscovery, which is responsible  for only this SceneType. So the SceneDiscoveryServiceManager also creates an own SceneDiscoveryService for each scene-type and registers it by the framework.

It is a little bit complicated. We hope you understand our explanation.